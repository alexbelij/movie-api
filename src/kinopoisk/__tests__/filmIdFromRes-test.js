/* @flow */

import {__isTitleSimilarToQuery, __filterResults} from '../filmIdFromRes';

describe('kinopoisk/filmIdFromRes', () => {
  it('checks title and query for similarity', () => {
    expect(
      __isTitleSimilarToQuery(
        'Star Wars: Episode VII - The Force Awakens',
        'Star Wars: The Force Awakens',
      ),
    ).toBe(true);
  });

  it('filters results by title', () => {
    expect(
      __filterResults({title: 'foo'}, [
        null,
        undefined,
        {id: 1, title: 'foo', countries: [], year: 2016},
      ]),
    ).toMatchSnapshot();

    expect(
      __filterResults({title: 'Звёздные войны'}, [
        {id: 1, title: 'Звёздные войны', countries: [], year: 2015},
        {id: 2, title: 'Звездные войны', countries: [], year: 2016},
        {id: 3, title: 'Звёздная пыль', countries: [], year: 2014},
        {id: 4, title: 'Звёздный путь', countries: [], year: 2016},
      ]),
    ).toMatchSnapshot();

    expect(
      __filterResults({title: 'Star Wars: Episode VII - The Force Awakens'}, [
        {
          id: 1,
          title: 'Звёздные войны',
          originalTitle: 'Star Wars: Episode VI',
          countries: [],
          year: 2015,
        },
        {
          id: 2,
          title: 'Звёздные войны',
          originalTitle: 'Star Wars: Episode VII - The Force Awakens',
          countries: [],
          year: 2016,
        },
        {
          id: 3,
          title: 'Звёздные войны',
          originalTitle: 'Star Wars: Episode V',
          countries: [],
          year: 2014,
        },
        {
          id: 4,
          title: 'Звёздные войны',
          originalTitle: 'Star Wars: Episode IV',
          countries: [],
          year: 2016,
        },
      ]),
    ).toMatchSnapshot();
  });

  it('filters results by year', () => {
    expect(
      __filterResults({title: 'foo', year: 2016}, [
        {id: 1, title: 'foo', countries: [], year: 2015},
        {id: 2, title: 'foo', countries: [], year: 2016},
        {id: 3, title: 'foo', countries: [], year: 2014},
        {id: 4, title: 'foo', countries: [], year: 2016},
      ]),
    ).toMatchSnapshot();
  });

  it('filters results by countries', () => {
    expect(
      __filterResults({title: 'foo', countries: ['США', 'Россия']}, [
        {id: 1, title: 'foo', countries: ['США'], year: 2016},
        {
          id: 2,
          title: 'foo',
          countries: ['США', 'Россия', 'Германия'],
          year: 2016,
        },
        {id: 3, title: 'foo', countries: ['Россия'], year: 2016},
        {id: 4, title: 'foo', countries: ['США', 'Россия'], year: 2016},
      ]),
    ).toMatchSnapshot();
  });

  it('filters results by type (movie or tv show)', () => {
    expect(
      __filterResults({title: 'Игра престолов', year: 2011, isTvShow: true}, [
        {id: 1, title: 'Игра престолов', countries: [], year: 2011},
        {id: 2, title: 'Игра престолов (сериал)', countries: [], year: 2011},
        {id: 3, title: 'Игра престолов (сериал)', countries: [], year: 2012},
        {id: 4, title: 'foo', countries: [], year: 2011},
      ]),
    ).toMatchSnapshot();

    expect(
      __filterResults({title: 'Звёздные войны', year: 2016, isTvShow: false}, [
        {id: 1, title: 'Звёздные войны (сериал)', countries: [], year: 2016},
        {id: 2, title: 'Звёздные войны', countries: [], year: 2015},
        {id: 3, title: 'Звёздные войны', countries: [], year: 2016},
        {id: 4, title: 'foo', countries: [], year: 2011},
      ]),
    ).toMatchSnapshot();
  });
});
