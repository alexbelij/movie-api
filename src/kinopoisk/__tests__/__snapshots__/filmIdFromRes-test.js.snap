// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`kinopoisk/filmIdFromRes filters results by countries 1`] = `
Array [
  Object {
    "countries": Array [
      "США",
      "Россия",
      "Германия",
    ],
    "id": 2,
    "title": "foo",
    "year": 2016,
  },
  Object {
    "countries": Array [
      "США",
      "Россия",
    ],
    "id": 4,
    "title": "foo",
    "year": 2016,
  },
]
`;

exports[`kinopoisk/filmIdFromRes filters results by title 1`] = `
Array [
  Object {
    "countries": Array [],
    "id": 1,
    "title": "foo",
    "year": 2016,
  },
]
`;

exports[`kinopoisk/filmIdFromRes filters results by title 2`] = `
Array [
  Object {
    "countries": Array [],
    "id": 1,
    "title": "Звёздные войны",
    "year": 2015,
  },
  Object {
    "countries": Array [],
    "id": 2,
    "title": "Звездные войны",
    "year": 2016,
  },
]
`;

exports[`kinopoisk/filmIdFromRes filters results by title 3`] = `
Array [
  Object {
    "countries": Array [],
    "id": 2,
    "originalTitle": "Star Wars: Episode VII - The Force Awakens",
    "title": "Звёздные войны",
    "year": 2016,
  },
]
`;

exports[`kinopoisk/filmIdFromRes filters results by type (movie or tv show) 1`] = `
Array [
  Object {
    "countries": Array [],
    "id": 2,
    "title": "Игра престолов (сериал)",
    "year": 2011,
  },
]
`;

exports[`kinopoisk/filmIdFromRes filters results by type (movie or tv show) 2`] = `
Array [
  Object {
    "countries": Array [],
    "id": 3,
    "title": "Звёздные войны",
    "year": 2016,
  },
]
`;

exports[`kinopoisk/filmIdFromRes filters results by year 1`] = `
Array [
  Object {
    "countries": Array [],
    "id": 2,
    "title": "foo",
    "year": 2016,
  },
  Object {
    "countries": Array [],
    "id": 4,
    "title": "foo",
    "year": 2016,
  },
]
`;
